### ISSUE TITLE
Logits vs probs when computing NLL?

### ISSUE THREAD

**Issue by najwalb** [cvignac/DiGress#14]
In the KL prior and L_{t-1} terms, you use the log of the probabilities to compute the kl divergence. See for example [here](https://github.com/cvignac/DiGress/blob/39ea947ea284ddf0e491f400865b9cedb3b0220e/src/diffusion_model_discrete.py#L316). Why do you use logits instead of the probabilities directly? And if you're taking the log of the model's probability, shouldn't you do the same for whatever you're comparing too (e.g. limit_dist in kl_prior or true posterior for L_{t-1} terms)?

**Comment by najwalb** [comment]
I found the answer, the use of log is required by the implementation of pytorch's [kl_div](https://pytorch.org/docs/stable/generated/torch.nn.functional.kl_div.html). I will close this issue now.

**Comment by cvignac** [comment]
Yes, it's a good idea to add some small values in the probs. In MiDi (https://github.com/cvignac/MiDi) we use 1e-7, but a large range of values is probably possible.

Clement

**Comment by najwalb** [comment]
Cool thanks. I will close the issue now.
